lib/segment/src/index/field_index/histogram.rs

```diff
@@ -179,6 +176,79 @@ impl<T: Numericable> Histogram<T> {
         Unbounded
     }
 
+    /// Infers boundaries for bucket of given size and starting point, skipping the first
+    /// border next to the starting point.
+    ///
+    /// E.g:
+    ///                                starts counting in this border
+    ///                                    ---> v
+    /// 1, 2, 3, 4, 5, 5, 5, 5, 6, 7, 8, 9, 10, 11, 12, 13
+    /// b           [                    b      b       b
+    ///           from
+    pub fn get_range_by_size_excluding(&self, from: Bound<T>, range_size: usize) -> Bound<T> {
```

Comment:
Is there a specific reason why we need to involve `histogram` into range reading?

AFAIK, it should be enough to use `range` of a btree (which is a DoubleEndedIterator). I would prefer to avoid using histogram for anything apart from it's original purpose

Reply:
💡 I actually didn't consider to use the `Iterator` part of the `range` function as a streaming source. It's a good idea, let me try to do that instead.